sdk/core/Azure.Core/tests/TransportFunctionalTests.cs

```diff
@@ -94,6 +94,35 @@ public async Task SettingHeaderOverridesDefaultContentLength()
             Assert.AreEqual(50, contentLength);
         }
 
+        [Test]
+        public async Task CanSetContentLenghtOverMaxInt()
+        {
+            long contentLength = 0;
+            using TestServer testServer = new TestServer(
+                context =>
+                {
+                    contentLength = context.Request.ContentLength.Value;
+                });
+            var maxValue = 2147483592;
```

Comment:
nit - make this int.MaxValue + 1 just to validate we are not capped by int anymore?